### DISCUSSION TITLE
Help needed

### DISCUSSION THREAD

**Discussion by ShresthVaibhav** [PaiGramTeam/PaiGram#222] [General]
I would like to ask assistance in making a repo related to playercard... Which would include mongo in place of redis and MySQL.... And will use saved genshin uid to get data from enka and make player card... It would be a great help from your side... Please feel free to decline. My telegram username - @SILVER_KING in case you agree to help.  I feel the ranking system of artifacts and weapon is great so i wanted to add that feature ... But i have no knowledge of ptb ... I have knowledge of telethon SQL and mongodb . Thanks for time

**Comment by luoshuijs** [comment]
We're glad you found our project helpful. Please feel free to dig through the code base to find something useful for your project. But for your technical background (as you mentioned), you may need more time than you expect to get familiar with this code base. This project uses not only basic platform interaction (PTB), but also more technical stacks such as web crawlers, HTML, etc. Game cards, artifacts, and weapon functions all require these technology stacks. In addition, these functions are highly dependent on each other (i.e. our modules cannot be freed from content coupling) and may have bugs that do not exist during migration. If your bots are still used in public groups, you must defend against flooding attacks and use Redis caching to improve performance and prevent host server overload. Therefore it is difficult for us to offer you more help. By the way, we already plan to add multilingual support to this project, and you are more than welcome to contribute.

**Reply by ShresthVaibhav** [reply]
Apologies for the delayed response! I've successfully created my own bot (@cocogoatxbot) with the assistance of enkanetwork and the hoyolab library. While it might not be as advanced in the hoyolab aspect compared to (@PaimonMasterBot), I'm actively enhancing it. Currently, it only supports English, but Enka cards can be generated in other languages. Thank you for suggesting ideas and libraries. If you ever need assistance, feel free to DM me at (@SILVER_KING).
